server/src/test/java/org/elasticsearch/action/admin/cluster/snapshots/status/TransportSnapshotsStatusActionTests.java

```diff
@@ -200,4 +204,142 @@ public void onFailure(Exception e) {
+                    Map.of(
+                        snapshot,
+                        Map.of(
+                            shardId2,
+                            new SnapshotIndexShardStatus(
+                                new ShardId(indexName, indexUuid, 2),
+                                SnapshotIndexShardStage.DONE,
+                                new SnapshotStats(nowMsecs, 0, 1, 1, 1, eightKb, eightKb, eightKb),
+                                "nodeId2",
+                                null
+                            )
+                        )
+                    )
+                )
+            ),
+            List.of()
+        );
+
+        final Consumer<SnapshotsStatusResponse> verifyResponse = rsp -> {
+            assertNotNull(rsp);
+            final var snapshotStatuses = rsp.getSnapshots();
+            assertNotNull(snapshotStatuses);
+            assertEquals(
+                "expected 1 snapshot status, got " + snapshotStatuses.size() + ": " + snapshotStatuses,
+                1,
+                snapshotStatuses.size()
+            );
+            final var snapshotStatus = snapshotStatuses.getFirst();
+            assertEquals(SnapshotsInProgress.State.STARTED, snapshotStatus.getState());
+            final var shardStats = snapshotStatus.getShardsStats();
+            assertNotNull("expected non-null shard stats for SnapshotStatus: " + snapshotStatus, shardStats);
+            assertEquals(new SnapshotShardsStats(0, 1 /* started */, 0, 2 /* done */, 0, 3 /* total */), shardStats);
+            final var totalStats = snapshotStatus.getStats();
+            assertNotNull("expected non-null total stats for SnapshotStatus: " + snapshotStatus, snapshotStatus);
+            assertEquals("expected total file count to be 1 in the stats: " + totalStats, 1, totalStats.getTotalFileCount());
+            assertEquals("expected total size to be " + eightKb + " in the stats: " + totalStats, eightKb, totalStats.getTotalSize());
+            final var snapshotStatusIndices = snapshotStatus.getIndices();
+            assertNotNull("expected a non-null map from getIndices() from SnapshotStatus: " + snapshotStatus, snapshotStatusIndices);
+            final var snapshotIndexStatus = snapshotStatusIndices.get(indexName);
+            assertNotNull(
+                "no entry for indexName [" + indexName + "] found in snapshotStatusIndices: " + snapshotStatusIndices,
+                snapshotIndexStatus
+            );
+            final var shardMap = snapshotIndexStatus.getShards();
+            assertNotNull("expected a non-null shard map for SnapshotIndexStatus: " + snapshotIndexStatus, shardMap);
+            final var shard0Entry = shardMap.get(0);
+            assertNotNull("no entry for shard 0 found in indexName [" + indexName + "] shardMap: " + shardMap, shard0Entry);
+            assertNotNull("expected a description string for shard 0 with missing stats from node0", shard0Entry.getDescription());
```

Comment:
You've verified that the description is set, but not what the stats contain. Could you check that part of the description matches part of the expected message? And that the stats are 0 (or whatever the defaults are)?

And can you verify that a normal shard snapshot (no node departure) does _not_ have a description?